### DISCUSSION TITLE
Ansible playbook in the root directory is getting skipped

### DISCUSSION THREAD

**Discussion by s-m-p** [super-linter/super-linter#5785] [Q&A]
Hi all. I've noticed that Ansible playbooks in the root directory of the repo are being skipped. I first noticed this using the Github Action, but have since been able to reproduce it running super-linter outside Github Actions in both the slim and full images. Most recently, I've used this command:
`sudo docker run -e RUN_LOCAL=true -e USE_FIND_ALGORITHM=true -e DEFAULT_BRANCH=main -e VALIDATE_ANSIBLE=true -e ANSIBLE_CONFIG_FILE=.ansible-lint -e LINTER_RULES_PATH=. -e LOG_LEVEL=DEBUG -v $(pwd):/tmp/lint ghcr.io/super-linter/super-linter:v6.6.0`

I am executing this command from a directory that contains two files:
- the `.ansible-lint` file
- an Ansible playbook file named `inv_check.yml'

When I look at the debug output, I notice this sequence of seeming-relevant log entries. This seems to indicate that the playbook file was found:
`2024-06-17 19:48:56 [DEBUG]   Categorizing the following files: /tmp/lint/.ansible-lint /tmp/lint/inv_check.yml /tmp/lint/test.sh /tmp/lint /tmp/lint`

Super-linter categorizes the playbook file:
`2024-06-17 19:48:56 [DEBUG]   FILE: /tmp/lint/inv_check.yml, FILE_TYPE: yml, BASE_FILE: inv_check.yml, FILE_DIR_NAME: /tmp/lint`

Super-linter did not include the playbook file in the list of to lint for Ansible:
`2024-06-17 19:48:56 [DEBUG]   /tmp/tmp.m5n7plnfkI/file-array-ANSIBLE doesn't exist. Skip loading the list of files and directories to lint for ANSIBLE`
`2024-06-17 19:48:56 [DEBUG]   There are no items to lint for ANSIBLE`

I'm no expert, but I've been studying the the code and it seems like this file test must have failed:
[https://github.deere.com/super-linter/super-linter/blob/56576d491db07c7236b445ab09991ca49d12b0c6/lib/functions/buildFileList.sh#L297](url)

But when I move the playbook to a subdirectory `playbooks` and include `ANSIBLE_DIRECTORY=playbooks` in the docker command, the playbook file is linted. I was thinking the issue might be that the test `[[ -d "${FILE}" ]]` must be false because ${FILE} is not a directory, but it seems like that test would always fail. But like I said before, it seems to work when I move the playbooks to a subdirectory.

Is there another enviroment variable(s) I should set when the Ansible playbooks are in the root of the repo?

**Comment by ferrarimarco** [comment]
The default value of `ANSIBLE_DIRECTORY` is `/ansible`. If you have your Ansible stuff in a directory that's not `/ansible`, you should set `ANSIBLE_DIRECTORY` as you described.

So, this is working as intended :)

**Reply by s-m-p** [reply]
The README makes this statement about `ANSIBLE_DIRECTORY`:
> Set to . to use the top-level of the DEFAULT_WORKSPACE.

But when I set it to `.`, super-linter doesn't find the playbook file.

**Reply by ferrarimarco** [reply]
Can you please open a new issue? We might need to dig deeper to check if it's a bug :)

**Reply by s-m-p** [reply]
Absolutely. I just wanted to make sure I wasn't missing something obvious before I did that, as I'm not very experienced with ansible or linting. Thanks!

**Comment by s-m-p** [comment]
I have confirmed the bug is fixed in the latest image. Thank you for such a quick response!

**Reply by ferrarimarco** [reply]
Thanks for the update!
